src/transform.nit

```diff
@@ -285,6 +285,57 @@ redef class AForExpr
 	end
 end
 
+redef class AWithExpr
+	# is replaced with a do/end and injected calls to `start` and `finish`
+	#
+	# Basically, the following
+	#
+	# ~~~nitish
+	# with expr do
+	#   block
+	# end label l
+	# ~~~
+	#
+	# is transformed into
+	#
+	# ~~~nitish
+	# var x = expr
+	# do
+	#   x.start
+	#   block
+	# end label l
+	# x.finish
+	# ~~~
+	#
+	# The point is that `finish` is called even if the block is escaped.
```

Comment:
Because the state or the semantic of the returned object can vary a lot and very complex, the simple way is just to handle what you need at the begin of the block.

~~~
with f = new FileWriter.open(path) then
    if f.has_errors then
        # things
        break
    end
    # exec only on success
end
~~~